app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -766,9 +779,10 @@ private void showSaveErrorAndExit() {
      *                        non-null
      * @param headless        Disables GUI warnings and lets answers that
      *                        violate constraints be saved.
+     * @param ignoreResult    If set, task's result will be ignored.
      */
     private void saveDataToDisk(boolean exit, boolean complete, String updatedSaveName,
-                                boolean headless) {
+                                boolean headless, boolean ignoreResult) {
```

Comment:
Ohh this is meant to be another layer of check. Like if contentProvider doesn't return info about file size then this would prevent us.

Reply:
can you remind me why do we need to ignore the save task result at all  and how does it serve as a second check ?

Reply:
So if the file size check fails, then there's another check at widget level(which is incorrect as of now btw since it's using unscaled file). 
The widget level check was causing a crash when it was done in incomplete form, since in incomplete form any changes to attachments are automatically saved to the disk at that time only. The crash was happening because on successfully completing the save task we were refereshing the screen(and there was a race condition with some other thread). So I had to ignore the result of save task to prevent that.

Reply:
is it possible to get more info about that crash ? If possible, I would like to avoid this design that routes a flag through save task in order to avoid a certain condition later in code path. And I am also curious what is it about second check that's different from first check that makes first check susceptible to failure but not the second check.

Reply:
> And I am also curious what is it about second check that's different from first check that makes first check susceptible to failure but not the second check.

So, the first check that I'm doing makes use of contentURI https://github.com/dimagi/commcare-android/pull/2421/files#diff-3499d048c2232bdd72d634a2a51d63ad8a1b804e3039654f48f7b4044ad1d5bbR726-R733 and it may not work always. The second check happens when the files is copied to the scoped storage and uses `file.length` which will always work. 

Another thing that I missed earlier was that the content URI will be of unscaled image. But it'll work for other media. 

> If possible, I would like to avoid this design that routes a flag through save task in order to avoid a certain condition later in code path.

I think that's possible. 
So I guess I'll close this PR, it already has a lot of unwanted commits, and then I create another one where I'll first try to use contentURI to validate file size and then before setting the answer, I'll check the file size using the file Path. And if it fails, we can show the error dialog to the user.

Reply:
yeah that makes sense to me. thanks!